lib/p2p/Peer.ts

```diff
@@ -264,6 +268,9 @@ class Peer extends EventEmitter {
       const onConnect = () => {
         this.connectTime = Date.now();
         this.connected = true;
+
+        this.bindSocket(socket);
+
```

Comment:
Are we sure there's no chance of events being called after connection, and prior to this binding?